eclair-core/src/main/scala/fr/acinq/eclair/payment/send/PaymentInitiator.scala

```diff
@@ -330,49 +327,44 @@ object PaymentInitiator {
-   * @param recipientAmount       amount that should be received by the final recipient (usually from a Bolt 11 invoice).
-   *                              This amount may be split between multiple requests if using MPP.
-   * @param invoice               Bolt 11 invoice.
-   * @param route                 route to use to reach either the final recipient or the first trampoline node.
-   * @param externalId            (optional) externally-controlled identifier (to reconcile between application DB and eclair DB).
-   * @param parentId              id of the whole payment. When manually sending a multi-part payment, you need to make
-   *                              sure all partial payments use the same parentId. If not provided, a random parentId will
-   *                              be generated that can be used for the remaining partial payments.
-   * @param trampolineSecret      if trampoline is used, this is a secret to protect the payment to the first trampoline
-   *                              node against probing. When manually sending a multi-part payment, you need to make sure
-   *                              all partial payments use the same trampolineSecret.
-   * @param trampolineFees        if trampoline is used, fees for the first trampoline node. This value must be the same
-   *                              for all partial payments in the set.
-   * @param trampolineExpiryDelta if trampoline is used, expiry delta for the first trampoline node. This value must be
-   *                              the same for all partial payments in the set.
-   * @param trampolineNodes       if trampoline is used, list of trampoline nodes to use (we currently support only a
-   *                              single trampoline node).
+   * @param recipientAmount amount that should be received by the final recipient (usually from a Bolt 11 invoice).
+   *                        This amount may be split between multiple requests if using MPP.
+   * @param invoice         Bolt 11 invoice.
+   * @param route           route to use to reach either the final recipient or the trampoline node.
+   * @param externalId      (optional) externally-controlled identifier (to reconcile between application DB and eclair DB).
+   * @param parentId        id of the whole payment. When manually sending a multi-part payment, you need to make
+   *                        sure all partial payments use the same parentId. If not provided, a random parentId will
+   *                        be generated that can be used for the remaining partial payments.
+   * @param trampoline_opt  if trampoline is used, this field must be provided. When manually sending a multi-part
+   *                        payment, you need to make sure all partial payments share the same values.
    */
-  case class SendPaymentToRoute(amount: MilliSatoshi,
-                                recipientAmount: MilliSatoshi,
+  case class SendPaymentToRoute(recipientAmount: MilliSatoshi,
```

Comment:
This is a bit hard to follow, I agree. But the `recipient`s are actually created in response to the `PaymentInitiator`'s `Send*`.
We should think of the `PaymentInitiator`'s commands as the external-facing API (it directly takes inputs from eclair's HTTP API), and their job is to create internal objects (such as `Recipient`) and send internal commands (to start payment FSMs).

We could at some point move that to `Eclair.scala` to remove a layer, but it would make it quite big, so it would deserve a slightly larger refactoring.

Reply:
Fair enough!